CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+See README.md for instruction on how to get, build, and test the source.  Once you have made your changes:
+
+
+
+*   Create a pull request (PR) with your patch.
+*   Make sure the PR description clearly identified the problem, explains the solution, and references the issue if applicable.
+*   Watch the continuous builds to make sure they pass.
+*   Request a code review.
+
+The reviewer can either approve your PR or request changes.  If changes are requested:
+
+
+
+*   Please add new commits to your branch, instead of amending your commit.  Adding new commits makes it easier for the reviewer to see what has changed since the last review.
+*   Once you are ready for another round of reviews, add a comment at the bottom, such as "Ready for review" or "Please take a look" (or "PTAL").   This explicit handoff is useful when responding with multiple small commits.
+
+After the PR has been reviewed it is the job of the reviewer to merge the PR.  Instructions for this are given below.
+
+
+## For maintainers: Reviewing a PR
+
+The formal code reviews are done on GitHub.  Reviewers are to look for all of the usual things:
+
+
+
+*   Coding style follows the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
+*   Identify potential functional problems.
+*   Identify code duplication.
+*   Ensure the unit tests have enough coverage.
+
+When looking for functional problems, there are some common problems reviewers should pay particular attention to:
+
+
+
+*   Does the code work for both Shader (Vulkan and OpenGL) and Kernel (OpenCL) scenarios?  The respective SPIR-V dialects are slightly different.
+*   Changes are made to a container while iterating through it.  You have to be careful that iterators are not invalidated or that elements are not skipped.
+*   C++11 and VS2013.  We generally assume that we have a C++11 compliant compiler.  However, on Windows, we still support Visual Studio 2013, which is not fully C++11 compliant. See [here](https://msdn.microsoft.com/en-us/library/hh567368.aspx).  In particular, note that it does not provide default move-constructors or move-assignments for classes.  In general, r-value references do not work the way you might assume they do.
+*   For SPIR-V transforms: The module is changed, but the analyses are not updated.  For example, a new instruction is added, but the def-use manager is not updated.  Later on, it is possible that the def-use manager will be used, and give wrong results.
+
+
+## For maintainers: Merging a PR
+
+We intend to maintain a linear history on the GitHub master branch, and the build and its tests should pass at each commit in that history.  A linear always-working history is easier understand and it easier to bisect in case we want to find which commit introduced a bug.
```

Comment:
"and it easier to bisect" --> "and easier to bisect"

Reply:
done